api/controllers/grades.go

```diff
@@ -6,283 +6,202 @@ import (
-			pipeline = mongo.Pipeline{courseMatch, lookupSectionsStage, unwindSectionsStage, projectGradeDistributionStage, unwindGradeDistributionStage, groupGradesStage, sortGradesStage, sumGradesStage, groupGradeDistributionStage}
-
-		case prefix != "" && number != "" && section_number != "" && !professor:
-			// Filter on Course then Section
-			collection = courseCollection
-			courseMatch := bson.D{{Key: "$match", Value: bson.M{"subject_prefix": prefix, "course_number": number}}}
-			sectionMatch := bson.D{{Key: "$match", Value: bson.M{"sections.section_number": section_number}}}
-			pipeline = mongo.Pipeline{courseMatch, lookupSectionsStage, unwindSectionsStage, sectionMatch, projectGradeDistributionStage, unwindGradeDistributionStage, groupGradesStage, sortGradesStage, sumGradesStage, groupGradeDistributionStage}
-
-		case prefix == "" && number == "" && section_number == "" && professor:
-			// Filter on Professor
-			collection = professorCollection
-
-			// Build professorMatch
-			if last_name == "" {
-				professorMatch = bson.D{{Key: "$match", Value: bson.M{"first_name": first_name}}}
-			} else if first_name == "" {
-				professorMatch = bson.D{{Key: "$match", Value: bson.M{"last_name": last_name}}}
-			} else {
-				professorMatch = bson.D{{Key: "$match", Value: bson.M{"first_name": first_name, "last_name": last_name}}}
-			}
-
-			// Build grades pipeline
-			pipeline = mongo.Pipeline{professorMatch, lookupSectionsStage, unwindSectionsStage, projectGradeDistributionStage, unwindGradeDistributionStage, groupGradesStage, sortGradesStage, sumGradesStage, groupGradeDistributionStage}
-
-		case prefix != "" && professor:
-			// Filter on Section by Matching Course and Professor IDs
-
-			// Here we get the valid course ids and professor ids
-			// and then we perform the grades aggregation against the sections collection,
-			// matching on the course_reference and professor
-
-			var profIDs []primitive.ObjectID
-			var courseIDs []primitive.ObjectID
-
-			collection = sectionCollection
-
-			// Find valid professor ids
-			if last_name == "" {
-				professorFind = bson.D{{Key: "first_name", Value: first_name}}
-			} else if first_name == "" {
-				professorFind = bson.D{{Key: "last_name", Value: last_name}}
-			} else {
-				professorFind = bson.D{{Key: "first_name", Value: first_name}, {Key: "last_name", Value: last_name}}
-			}
-
-			cursor, err = professorCollection.Find(ctx, professorFind)
-			if err != nil {
-				c.JSON(http.StatusInternalServerError, responses.GradeResponse{Status: http.StatusInternalServerError, Message: "error", Data: err.Error()})
-				return
-			}
-			if err = cursor.All(ctx, &results); err != nil {
-				panic(err)
-			}
 
-			for _, prof := range results {
-				profID := prof["_id"].(primitive.ObjectID)
-				profIDs = append(profIDs, profID)
-			}
+		pipeline = append(pipeline, bson.D{{Key: "$match", Value: sectionMatch}})
```

Comment:
This chunk doesn't produce valid queries either, and misses most of the pipeline.